rslint_core/src/groups/errors/valid_typeof.rs

```diff
@@ -0,0 +1,120 @@
+use crate::rule_prelude::*;
+use ast::{BinExpr, Expr, Literal, UnaryExpr};
+
+declare_lint! {
+    /**
+    Enforce the use of valid string literals in a `typeof` comparison.
+
+    ## Invalid Code Examples
+    ```ignore
+    typeof foo === "strnig"
+    typeof foo == "undefimed"
+    typeof bar != "nunber"
+    typeof bar !== "fucntion"
+    ```
+    */
+    #[derive(Default)]
+    ValidTypeof,
+    errors,
+    "valid-typeof",
+
+    #[serde(default)]
```

Comment:
I would rather this be put on the struct itself on top of the derive default to be consistent